include/ensmallen_bits/aug_lagrangian/aug_lagrangian.hpp

```diff
@@ -49,13 +48,24 @@ class AugLagrangian
    *     class.
    * @param function The function to optimize.
    * @param coordinates Output matrix to store the optimized coordinates in.
+   * @param penaltyThresholdFactor When the penalty threshold is updated set
+   *    the penalty threshold to the penalty multplied by this factor. The
+   *    default value of 0.25 is is taken from Burer and Monteiro (2002).
+   * @param sigmaUpdateFactor When sigma is updated  multiply sigma by this
+   *    value. The default value of 10 is taken from Burer and Monteiro (2002).
    * @param maxIterations Maximum number of iterations of the Augmented
    *     Lagrangian algorithm.  0 indicates no maximum.
+   * @param internalMaxIterations Maximum number of iterations of L-BFGS
+   *    iterations internal to the Augmented Lagrangian algorithm.
+   *    0 indicates no maximum.
```

Comment:
We can actually expose all of the members of the internal L-BFGS optimizer by having a parameter `lbfgs` of type `const L_BFGS&`.  Then, a user can set the maximum iterations of the L-BFGS optimizer in two ways:

```
L_BFGS l(..., 1000 /* max iterations */, ...);
AugLagrangian aug(..., l);
```

Or they can do it like this:

```
AugLagrangian aug(...);
aug.LBFGS().MaxIterations() = 1000;
```

A nice benefit to that is they can also access all the other internal members of the L-BFGS optimizer.

Reply:
I'll remove my exposing of the LBFGS max iterations.

That first constructor which takes an LBFGS object doesn't currently exist though, shall I add it?

Reply:
Right, exactly, we'd need to add that as a constructor argument.  We can have a default too, so you can just modify the existing constructor and add `const L_BFGS& lbfgs = L_BFGS()`.  :+1: